### ISSUE TITLE
[Feature] Tag Blacklisting

### ISSUE THREAD

**Issue by SpedNSFW** [stashapp/stash#985]
Similar train of thought to #121 but instead of mapping the tag to another name, have a blacklist for tags. Some studios have stupid tags for their scenes like "HD", "Straight", "Pornstar", etc that you don't want to see.

My request is to add an interface where you can enter any tag you wish to be blacklisted and when you scrape a new scene that features those tags, they don't get added (or prompted to be added).

**Comment by gitgiggety** [comment]
@WithoutPants I wanna have a stab at this. But do you have any suggestions for it? Storing the blacklist in a new database table? Of storing it in the config file? Or a separate (txt) file containing a single term per line?
And suggestions for the UI part are welcome as well.

Especially as I do believe such a blacklist might get longish. I.e. might go to 50 or more terms I guess. So for example storing in the config file and adding to the Configuration tab of the settings (below the Scraping header) might not be a good idea.

**Comment by hyde231** [comment]
well, what is a central config file good for, if you don't consequently store the user config in there? 
just my 2 cents

**Comment by gitgiggety** [comment]
@hyde231 It's hard to disagree with you on the fact that user configuration should be stored as such. But for one the user configuration is only configured using the interface, so does it really matter where and how it is stored? And secondly a list of blacklisted tags, which I guess might contain tens if not over 100 items can also be described as a "database" of blacklisted tags, and not necessarily as configuration. So storing it in the database (or flat file as being a flat file database) might be a better solution.

Which obviously is also why I'm asking the question, before developing it one way while it's preferred to implement differently.

**Comment by WithoutPants** [comment]
I don't think it belongs in the database. I think adding it to the configuration file is fine for now. We can look at migrating it out later if we need to.

For UI presentation, I think a collapsible section for it would suit. I like the presentation of the blacklist in the tagger. 

We have a `Scrapers` tab already. Perhaps we should consider renaming it to `Scraping` and move the scraping related configuration to that page. The existing scrapers list can be put into a collapsible section.

**Comment by gitgiggety** [comment]
Thanks for the input. I'll probably split it then. First do the UI refactoring and afterwards adding the tags blacklisting. As both changes can for example be reviewed independent of each other.

**Comment by gitgiggety** [comment]
@WithoutPants I'm working on the part of moving the settings around. But the scraper settings are part of the "general configuration" in the GraphQL API as well (and there also being "interface" and "dlna" configuration). So I guess it would be best to split of scraping configuration as well? But this does mean an API breakage etc. So I just want to verify whether I can/must do this refactor as well, or whether I should try to build it without such a refactor / breaking the (backwards compatibility of the) API.

So in short these changes would be:
* define `ConfigScraping{Input,Result}`
* add queries & mutations for it
* make the necessary back-end changes etc
* remove these three fields from the `ConfigGeneral{Input,Result}`
